Swerve2022/src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -23,11 +24,10 @@
   private static final double          kModuleMaxAngularVelocity     = Drivetrain.kMaxAngularSpeed;
   private static final double          kModuleMaxAngularAcceleration = 2 * Math.PI;   // radians per second squared
 
-  private final MotorController        m_driveMotor;
-  private final MotorController        m_turningMotor;
+  private final WPI_TalonFX            m_driveMotor;
+  private final WPI_TalonFX            m_turningMotor;
 
-  private final Encoder                m_driveEncoder;
-  private final Encoder                m_turningEncoder;
+  private final CANCoder               m_turningCANCoder;
```

Comment:
Is there only supposed to be a CANCoder for turning and not for driving?

Reply:
Great question!

The driving motor turns the wheel on its axle in the driving direction. The Falcon internal (integrated) encoder can be used to measure distance traveled in the direction the wheel rolls. It does not need a separate CANcoder to measure rotation. The turn motor is also driven by a Falcon with an internal encoder, but the internal encoder is not useful for measuring the direction the wheel is pointing, so a separate CANcoder is installed on that output shaft.

The CANcoder measures absolute position (in this case the angle the wheel is pointing) for the turn motor only.